docs/reference/specifications/proposal/rfc-grpc-custom-name-resolver.md

```diff
@@ -0,0 +1,157 @@
+application with S2S policy e.g. istio, envoy, consul etc. In such deployments the current gRPC connection was
+breaking because the proxies need the destination domain i.e. service name in order to route the traffic.
+
+For such cases the gRPC core libs support few alternative resolver* also expose required interface to build custom one
+you can find more details in the document referenced below
+
+### Reference
+
+* [Custom Name Resolution](https://grpc.io/docs/guides/custom-name-resolution/)
+* [Java Client](https://grpc.github.io/grpc-java/javadoc/io/grpc/ManagedChannelBuilder.html#forTarget(java.lang.String))
+* [Golang](https://pkg.go.dev/google.golang.org/grpc#NewClient)
+
+A sample workflow of a deployment with proxy sidecar using `in-process` mode (for `rpc` the process is same)
+
+```mermaid
+sequenceDiagram
+    participant flagd-sync.service
+    participant proxy-sidecar-agent
+    participant flagd-provider
+    participant application
+
+
+    flagd-provider-->proxy-sidecar-agent: Route traffic to backend sync service based on policy
+    Note left of flagd-provider: In most cases the connection made to <br> `localhost` data plane port <br> e.g. localhost:9211
+
+    flagd-provider-->flagd-sync.service: Get realtime flag update over gRPC stream
+    Note left of proxy-sidecar-agent: Based on the host header i.e. authority <br> the tarrfic was then routed <br> backend service.
+    loop
+        flagd-provider->>flagd-provider: in-memory cache
+    end
+    application-->>flagd-provider: Check the state of a feature flag
+    flagd-provider-->>application: Get the feature flag from in-memory cache <br/> run the evaluation logic and return final state
+```
+
+**Note:** There is small variation in supported alternative resolver e.g. java support `zooKeeper`
+
+## Motivation
+
+The main motivation is to support complex deployments with a generic custom name resolver using the interface
+provided by gRPC core*.
+
+**Note**: As of now only `java` and `golang` has the required interface to create custom resolver
+
+## Detailed design
+
+The idea is to
+
+* create custom name resolver for `java` and `golang` based on the example [provided](https://grpc.io/docs/guides/custom-name-resolution/#language-support)
+* and allow a new config option to pass the [target](https://grpc.io/docs/guides/custom-name-resolution/#life-of-a-target-string)
+string
+
+### Target String Pattern*
+
+There is no restriction on naming but the string most comply with below standard
+
+```text
+scheme://authority/endpoint_name
+```
+
+#### Scheme Suggestions
+
+We can choose our own scheme considering it's not used in gRPC core
+
+| Scheme             | Comment | Status             |
+|--------------------|---------|--------------------|
+| **flagd://**       |         | Approved / Reject  |
+| **grpc-remote://** |         | Approved / Reject  |
+| **tbd**            |         | Approved / Reject  |
```

Comment:
Can you update this section to be less about your specific use case? Maybe just give an example scheme and what it might do in a hypothetical infrastructure.

Reply:
done replace the section with envoy sidecar example